docs/build-tools/proxy.md

```diff
@@ -0,0 +1,176 @@
+Coursier version, including custom repositories, so mirrors are really the only
+way to fix this.
+
+There are different ways to setup mirrors:
+
+### Via global properties file
+
+Create `mirror.properties` file with the contents:
+
+```
+central.from=https://repo1.maven.org/maven2
+central.to=http://exmaple.com:8080/nexus/content/groups/public
+```
+
+You need to replace the uri after `central.to=` with your private repository.
+
+The location of the mirror file varies depending on the operating system:
+
+- Windows:
+  `C:\Users\<user_name>\AppData\Roaming\Coursier\config\mirror.properties`
+- Linux: `~/.config/coursier/mirror.properties`
+- MacOS: `~/Library/Preferences/Coursier/mirror.properties`
+
+This solution will work in most cases for all previously mentioned usages of
+Coursier, which is especially important in the case of the Bloop server, which when
+started by Metals will not have any system properties forwarded.
+
+### Via environment variable
+
+You can set the environment variable `COURSIER_MIRRORS` with the location of the
+file containing the mirror definition. That environment variable will need to
+available either locally or globally for the Metals server. In case of the
+Visual Studio Code extension it means you will need to start it from command
+line with that variable in scope.
+
+### Via properties
+
+You can also use properties to specify the location of the mirror file, for
+example:
+
+```
+-Dcoursier.mirrors=~/.config/coursier/mirror.properties
+```
+
+This property needs to be added to the Metals server and Coursier invocation.
+However, this solution will not work for Bloop, because properties are not
+forwarded to the invocation of the Bloop server. You will need to use a
+different method for Bloop in this case or you can run the build server manually
+with everything specified separately.
+
+## Custom artifact repositories (Maven or Ivy resolvers)
+
+In case you need to add custom repositories to resolve Metals server artifacts
+you can use the `COURSIER_REPOSITORIES` environment variable. This will tell
+Coursier to try to download artifacts from your private artifact repository.
+This is also available as a setting in the Metals Visual Studio Code and coc-metals extensions.
+
+## Proxy settings
+
+In some cases, workspaces might require a proxy in order to resolve the needed
+artifacts. Depending on the way Metals server is started, proxy settings can be
+specified using properties inside a `.jvmopts` file and `metals.serverProperties`
+(for Visual Studio Code or coc-metals extensions) or via properties for Coursier and Metals
+invocations.
+
+However, because proxy properties might vary between workspaces and a Bloop
+server must work for multiple clients at the same time they are not forwarded to
+the Bloop itself. To make sure that Bloop uses the correct proxy settings, you
+can specify them in two ways:
+
+- via copying the correct `.jvmopts` file to the `~/.bloop` directory, which will
```

Comment:
What about just recommending this option? When the server is started, proxy settings should just work. I would then tell users that if they have different proxy settings per workspace, every time they run `bloop` on those will reapply the proxy settings as they are global. This shouldn't be a big deal in most cases because Bloop only uses these settings to download the compiler bridges and the semanticdb, which are usually downloaded the first time a compilation happens.

Reply:
Thanks! I left the single way to define proxy settings and added more details about it.